src/prefect/client/client.py

```diff
@@ -703,14 +704,35 @@ def _request(
         retries = requests.packages.urllib3.util.retry.Retry(
             total=retry_total,
             backoff_factor=1,
-            status_forcelist=[429, 500, 502, 503, 504],
+            status_forcelist=[500, 502, 503, 504],
             method_whitelist=["DELETE", "GET", "POST"],
         )
         session.mount("https://", requests.adapters.HTTPAdapter(max_retries=retries))
         response = self._send_request(
             session=session, method=method, url=url, params=params, headers=headers
         )
 
+        # custom logic when encountering an API rate limit:
+        # each time we encounter a rate limit, we sleep for
+        # 3 minutes + random amount, where the random amount
+        # is uniformly sampled from (0, 10 * 2 ** rate_limit_counter)
+        # up to (0, 640), at which point an error is raised if the limit
+        # is still being hit
+        rate_limited = response.status_code == 429
+        iter_count = 1
+        while rate_limited:
+            jitter = random.random() * 10 * (2 ** iter_count)
+            naptime = 3 * 60 + jitter  # 3 + e min sleep
+            self.logger.debug("Rate limit encountered; sleeping...")
```

Comment:
It might be nice to include the seconds in the log; no strong feelings here
```suggestion
            self.logger.debug(f"Rate limit encountered; sleeping for {naptime}s...")
```